### PR TITLE
fix: return early for out-of-bounds mmap reads

### LINKED ISSUE #3446: Handle some corner case in `archive export`
**Describe the bug**

Not a big deal but giving an empty file to `forest-cli archive export` will result in a panic.

**To Reproduce** Steps to reproduce the behaviour:

1. Run `touch dummy.car`
2. Run `forest-cli archive export dummy.car`
3. See error

**Log output**

<details>
  <summary>Log Output</summary>
  
  ```
$ forest-cli archive export dummy.car
thread 'main' panicked at 'attempt to add with overflow', src/utils/io/mmap.rs:25:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  ```
</details>

**Expected behaviour**

We should handle this situation without crashing.

**Environment (please complete the following information):**

- OS: macOS
- Branch: main

**Other information and links**

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- return early if mmap read request is out-of-bounds.

## Reference issue to close (if applicable)

Closes #3446

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by aatifsyed — CHANGES_REQUESTED** [review]
I don't think this should be a shell script

File: scripts/tests/forest_tool_check.sh [hunk 1]
```diff
@@ -5,6 +5,12 @@
...
-FOREST_TOOL_PATH="forest-tool"
+source "$(dirname "$0")/harness.sh"
 
 "$FOREST_TOOL_PATH" state-migration actor-bundle
+
+# Exporting with an empty car should fail but not panic
+touch empty.car
+ERROR=$($FOREST_CLI_PATH archive export empty.car 2>&1 || true)
+assert_eq "$ERROR" "Error: input not recognized as any kind of CAR data (.car, .car.zst, .forest.car)"
+rm empty.car
```
Comment (aatifsyed) [inline scripts/tests/forest_tool_check.sh:16]: Can you write something like https://github.com/ChainSafe/forest/blob/62df2d8d608f2629664ad35e3e82fe3f930ee038/tests/import_snapshot_tests.rs?

```rust
touch("empty.car");
forest_cli().args(["archive export empty.car").assert().failure().stdout().contains("Error: input not recognized...")
```

Bonus points if you port the existing test and remove this file.

See also https://github.com/ChainSafe/forest/issues/2945

File: src/utils/io/mmap.rs [hunk 2]
```diff
@@ -22,6 +22,9 @@ impl Mmap {
 impl ReadAt for Mmap {
     fn read_at(&self, pos: u64, buf: &mut [u8]) -> io::Result<usize> {
         let start = pos as usize;
+        if start >= self.0.len() {
+            return Ok(0);
```
Comment (aatifsyed) [inline src/utils/io/mmap.rs:26]:
```suggestion
            // This matches the behaviour for seeking past the end of a file
            return Ok(0);
```

**Review by aatifsyed — APPROVED** [review]
Fantastic!

**Comment by lemmih** [comment]
Hmm, turns out that `state-migration actor-bundle` was in a separate script to bypass the 30s timeout.

**Comment by aatifsyed** [comment]
> Hmm, turns out that `state-migration actor-bundle` was in a separate script to bypass the 30s timeout.

I'd advocate for keeping it in rust, and segregating/configuring it via nextest

File: .config/nextest.toml [hunk 3]
```diff
@@ -1,2 +1,6 @@
 [profile.default]
 slow-timeout = { period = "10s", terminate-after = 3 }
+
+[[profile.default.overrides]]
+filter = 'test(state_migration_actor_bundle_runs)'
+slow-timeout = { period = "120s", terminate-after = 3 }
```
Comment (LesnyRumcajs) [inline .config/nextest.toml:6]: Why is it so slow? Can we make it faster? I'd very much prefer a regular cargo test to be speedy.

Comment (aatifsyed) [reply .config/nextest.toml:6]: Or alternatively, have it run in ci only using profiles: https://nexte.st/book/configuration.html#profiles

**Review by sudo-shashank — APPROVED** [review]

**Review by elmattic — APPROVED** [review]

File: .config/nextest.toml (hunk 3 above)
Comment (lemmih) [reply .config/nextest.toml:6]: I'll keep in the regular tests for now. Otherwise, it would be out of sight and never fixed. :)

**Pull request merged**